[BUG] Disabling ALTERNATIVE_MOVE_MENU breaks compilation

### Description

Disabling `ALTERNATIVE_MOVE_MENU` breaks compilation.

### Steps to reproduce

1. Use the [config](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/blob/b7d736eeab490563358726870fb945bc8ce49939/TFT/src/User/Configuration.h) from b7d736e, and comment out `ALTERNATIVE_MOVE_MENU`.
2. Compile
3. Observe errors:

```prolog
TFT/src/User/Menu/Mode.c: In function 'menuMode':
TFT/src/User/Menu/Mode.c:106:11: warning: unused variable 'nowEncoder' [-Wunused-variable]
   int16_t nowEncoder = encoderPosition = 0;
           ^~~~~~~~~~
TFT/src/User/Menu/Move.c: In function 'menuMove':
TFT/src/User/Menu/Move.c:251:32: error: 'XGCODE_UP' undeclared (first use in this function); did you mean 'XGCODE_DEC'?
                       storeCmd(XGCODE_UP, item_move_len[item_move_len_i]);  
                                ^~~~~~~~~
                                XGCODE_DEC
TFT/src/User/Menu/Move.c:251:32: note: each undeclared identifier is reported only once for each function it appears in
TFT/src/User/Menu/Move.c:254:32: error: 'YGCODE_UP' undeclared (first use in this function); did you mean 'XGCODE_UP'?
                       storeCmd(YGCODE_UP, item_move_len[item_move_len_i]);
                                ^~~~~~~~~
                                XGCODE_UP
TFT/src/User/Menu/Move.c:257:32: error: 'ZGCODE_UP' undeclared (first use in this function); did you mean 'YGCODE_UP'?
Compiling .pio/build/BIGTREE_TFT35_V3_0/src/src/User/Menu/PowerFailed.oi]);

Compiling .pio/build/BIGTREE_TFT35_V3_0/src/src/User/Menu/PreheatMenu.o
                                ^~~~~~~Compiling .pio/build/BIGTREE_TFT35_V3_0/src/src/User/Menu/Print.o
~~
                                YGCODE_UP
TFT/src/User/Menu/Move.c:268:32: error: 'XGCODE_DOWN' undeclared (first use in this function); did you mean 'XGCODE_DEC'?
                       storeCmd(XGCODE_DOWN, item_move_len[item_move_len_i]);  
                                ^~~~~~~~~~~
                                XGCODE_DEC
TFT/src/User/Menu/Move.c:271:32: error: 'YGCODE_DOWN' undeclared (first use in this function); did you mean 'XGCODE_DOWN'?
                       storeCmd(YGCODE_DOWN, item_move_len[item_move_len_i]);
                                ^~~~~~~~~~~
                                XGCODE_DOWN
compilation terminated due to -fmax-errors=5.
Compiling .pio/build/BIGTREE_TFT35_V3_0/src/src/User/Menu/Printing.o
```

**Expected behavior**
No compile errors

**Actual behavior**
Compile errors

### Hardware Variant
TFT35 3.0 and TFT35 E3 3.0

### TFT Firmware Version & Main Board Firmware details
b7d736e, TFT_35_V3.0.25.2 Mar 9 2020

### Additional Information

I can change the variables to `*_INC`/`*_DEC` to allow a successful compile, but then the move buttons no longer work.

--- a/TFT/src/User/API/UI/list_item.h
+++ b/TFT/src/User/API/UI/list_item.h
@@ -8,7 +8,7 @@
 #include "GUI.h"
 #include "menu.h"
 
-enum{
+typedef enum{
 ICONCHAR_BLANK = 0,
 ICONCHAR_NOZZLE,
 ICONCHAR_BED,

--- a/TFT/src/User/API/flashStore.c
+++ b/TFT/src/User/API/flashStore.c
@@ -3,7 +3,7 @@
 
 #define PARA_SIZE 256  //bytes
 #define TSC_SIGN  0x20190827 // DO NOT MODIFY
-#define PARA_SIGN 0x20200229 // If a new setting parameter is added, modify here and initialize the initial value in the "infoSettingsReset()" function
+#define PARA_SIGN 0x20200313 // If a new setting parameter is added, modify here and initialize the initial value in the "infoSettingsReset()" function
 
 extern u32 TSC_Para[7];        //
 extern SETTINGS infoSettings;  //
@@ -55,25 +55,25 @@ bool readStoredPara(void)
   }
   else
   {
-    infoSettings.baudrate         = byteToWord(data + (index += 4), 4);
-    infoSettings.language         = byteToWord(data + (index += 4), 4);
-    infoSettings.mode             = byteToWord(data + (index += 4), 4);
-    infoSettings.runout           = byteToWord(data + (index += 4), 4);
-    infoSettings.rotate_ui        = byteToWord(data + (index += 4), 4);
-    infoSettings.bg_color         = byteToWord(data + (index += 4), 4);
-    infoSettings.font_color       = byteToWord(data + (index += 4), 4);
-    infoSettings.silent           = byteToWord(data + (index += 4), 4);
-    infoSettings.auto_off         = byteToWord(data + (index += 4), 4);
-    infoSettings.terminalACK      = byteToWord(data + (index += 4), 4);
-    infoSettings.invert_xaxis     = byteToWord(data + (index += 4), 4);
-    infoSettings.invert_yaxis     = byteToWord(data + (index += 4), 4);
-    infoSettings.move_speed       = byteToWord(data + (index += 4), 4);
-    infoSettings.invert_zaxis     = byteToWord(data + (index += 4), 4);
-    infoSettings.send_start_gcode = byteToWord(data + (index += 4), 4);
-    infoSettings.send_end_gcode   = byteToWord(data + (index += 4), 4);
-    infoSettings.persistent_info  = byteToWord(data + (index += 4), 4);
-    infoSettings.file_listmode    = byteToWord(data + (index += 4), 4);
-    infoSettings.knob_led_color   = byteToWord(data + (index += 4), 4);
+    infoSettings.baudrate            = byteToWord(data + (index += 4), 4);
+    infoSettings.language            = byteToWord(data + (index += 4), 4);
+    infoSettings.mode                = byteToWord(data + (index += 4), 4);
+    infoSettings.runout              = byteToWord(data + (index += 4), 4);
+    infoSettings.rotate_ui           = byteToWord(data + (index += 4), 4);
+    infoSettings.bg_color            = byteToWord(data + (index += 4), 4);
+    infoSettings.font_color          = byteToWord(data + (index += 4), 4);
+    infoSettings.silent              = byteToWord(data + (index += 4), 4);
+    infoSettings.auto_off            = byteToWord(data + (index += 4), 4);
+    infoSettings.terminalACK         = byteToWord(data + (index += 4), 4);
+    infoSettings.invert_axis[X_AXIS] = byteToWord(data + (index += 4), 4);
+    infoSettings.invert_axis[Y_AXIS] = byteToWord(data + (index += 4), 4);
+    infoSettings.invert_axis[Z_AXIS] = byteToWord(data + (index += 4), 4);
+    infoSettings.move_speed          = byteToWord(data + (index += 4), 4);
+    infoSettings.send_start_gcode    = byteToWord(data + (index += 4), 4);
+    infoSettings.send_end_gcode      = byteToWord(data + (index += 4), 4);
+    infoSettings.persistent_info     = byteToWord(data + (index += 4), 4);
+    infoSettings.file_listmode       = byteToWord(data + (index += 4), 4);
+    infoSettings.knob_led_color      = byteToWord(data + (index += 4), 4);
   }
 
   return paraExist;
@@ -100,10 +100,10 @@ void storePara(void)
   wordToByte(infoSettings.silent,             data + (index += 4));
   wordToByte(infoSettings.auto_off,           data + (index += 4));
   wordToByte(infoSettings.terminalACK,        data + (index += 4));
-  wordToByte(infoSettings.invert_xaxis,       data + (index += 4));
-  wordToByte(infoSettings.invert_yaxis,       data + (index += 4));
+  wordToByte(infoSettings.invert_axis[X_AXIS],data + (index += 4));
+  wordToByte(infoSettings.invert_axis[Y_AXIS],data + (index += 4));
+  wordToByte(infoSettings.invert_axis[Z_AXIS],data + (index += 4));
   wordToByte(infoSettings.move_speed,         data + (index += 4));
-  wordToByte(infoSettings.invert_zaxis,       data + (index += 4));
   wordToByte(infoSettings.send_start_gcode,   data + (index += 4));
   wordToByte(infoSettings.send_end_gcode,     data + (index += 4));
   wordToByte(infoSettings.persistent_info,    data + (index += 4));

--- a/TFT/src/User/Configuration.h
+++ b/TFT/src/User/Configuration.h
@@ -163,7 +163,7 @@
 #ifdef AUTO_BED_LEVELING
   // Enable this will send "M500" after "G29" to store leveling value
   // and send "M420 S1" to enable leveling state after startup
-  #define AUTO_SAVE_LOAD_LEVELING_VALUE
+  //#define AUTO_SAVE_LOAD_LEVELING_VALUE
 #endif
 
 // Move to four corner points to Leveling manually (Point 1, Point 2, Point 3, Point 4)

--- a/TFT/src/User/Menu/FeatureSettings.c
+++ b/TFT/src/User/Menu/FeatureSettings.c
@@ -114,24 +114,24 @@ void updateFeatureSettings(uint8_t key_val)
     break;
       
     case SKEY_INVERT_X:
-    infoSettings.invert_xaxis = (infoSettings.invert_xaxis + 1) % TOGGLE_NUM;
-    settingPage[item_index].icon = toggleitem[infoSettings.invert_xaxis];
+    infoSettings.invert_axis[X_AXIS] = (infoSettings.invert_axis[X_AXIS] + 1) % TOGGLE_NUM;
+    settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[X_AXIS]];
     featureSettingsItems.items[key_val] = settingPage[item_index];
 
     menuDrawListItem(&featureSettingsItems.items[key_val], key_val);
     break;
 
     case SKEY_INVERT_Y:
-    infoSettings.invert_yaxis = (infoSettings.invert_yaxis + 1) % TOGGLE_NUM;
-    settingPage[item_index].icon = toggleitem[infoSettings.invert_yaxis];
+    infoSettings.invert_axis[Y_AXIS] = (infoSettings.invert_axis[Y_AXIS] + 1) % TOGGLE_NUM;
+    settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[Y_AXIS]];
     featureSettingsItems.items[key_val] = settingPage[item_index];
 
     menuDrawListItem(&featureSettingsItems.items[key_val], key_val);
     break;
 
     case SKEY_INVERT_Z:
-    infoSettings.invert_zaxis = (infoSettings.invert_zaxis + 1) % TOGGLE_NUM;
-    settingPage[item_index].icon = toggleitem[infoSettings.invert_zaxis];
+    infoSettings.invert_axis[Z_AXIS] = (infoSettings.invert_axis[Z_AXIS] + 1) % TOGGLE_NUM;
+    settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[Z_AXIS]];
     featureSettingsItems.items[key_val] = settingPage[item_index];
 
     menuDrawListItem(&featureSettingsItems.items[key_val], key_val);
@@ -228,17 +228,17 @@ void loadFeatureSettings(){
         break;
 
 		case SKEY_INVERT_X:
-        settingPage[item_index].icon = toggleitem[infoSettings.invert_xaxis];
+        settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[X_AXIS]];
         featureSettingsItems.items[i] = settingPage[item_index];
         break;
         
       case SKEY_INVERT_Y:
-        settingPage[item_index].icon = toggleitem[infoSettings.invert_yaxis];
+        settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[Y_AXIS]];
         featureSettingsItems.items[i] = settingPage[item_index];
         break;
 
       case SKEY_INVERT_Z:
-        settingPage[item_index].icon = toggleitem[infoSettings.invert_zaxis];
+        settingPage[item_index].icon = toggleitem[infoSettings.invert_axis[Z_AXIS]];
         featureSettingsItems.items[i] = settingPage[item_index];
         break;
 

--- a/TFT/src/User/Menu/Move.c
+++ b/TFT/src/User/Menu/Move.c
@@ -115,26 +115,38 @@ void menuMove(void)
       if(infoSettings.invert_xaxis == 1){
         moveItems.items[0].label.index = LABEL_X_DEC;
         moveItems.items[4].label.index = LABEL_X_INC;
+        x_axis_up   = XGCODE_DEC;
+        x_axis_down = XGCODE_INC;
       }
       else{
         moveItems.items[0].label.index = LABEL_X_INC;
         moveItems.items[4].label.index = LABEL_X_DEC;
+        x_axis_up   = XGCODE_INC;
+        x_axis_down = XGCODE_DEC;
       }
       if(infoSettings.invert_yaxis == 1){
         moveItems.items[1].label.index = LABEL_Y_DEC;
         moveItems.items[5].label.index = LABEL_Y_INC;
+        y_axis_up   = YGCODE_DEC;
+        y_axis_down = YGCODE_INC;
       }
       else{
         moveItems.items[1].label.index = LABEL_Y_INC;
         moveItems.items[5].label.index = LABEL_Y_DEC;
+        y_axis_up   = YGCODE_INC;
+        y_axis_down = YGCODE_DEC;
       }
         if(infoSettings.invert_zaxis == 1){
         moveItems.items[2].label.index = LABEL_Z_DEC;
         moveItems.items[6].label.index = LABEL_Z_INC;
+        z_axis_up   = ZGCODE_DEC;
+        z_axis_down = ZGCODE_INC;
       }
       else{
         moveItems.items[2].label.index = LABEL_Z_INC;
         moveItems.items[6].label.index = LABEL_Z_DEC;
+        z_axis_up   = ZGCODE_INC;
+        z_axis_down = ZGCODE_DEC;
       }
   #endif
 
@@ -240,7 +252,7 @@ void menuMove(void)
         case KEY_ICON_7: infoMenu.cur--; break;
         
         default:
-              #if LCD_ENCODER_SUPPORT && !MENU_LIST_MODE
+              #if LCD_ENCODER_SUPPORT
                 if(encoderPosition)
                 {
                   if(encoderPosition > 0)
@@ -248,13 +260,13 @@ void menuMove(void)
                     switch(a)
                     {
                     case X_AXIS:
-                      storeCmd(XGCODE_UP, item_move_len[item_move_len_i]);  
+                      storeCmd(x_axis_up, item_move_len[item_move_len_i]);  
                       break;
                     case Y_AXIS:
-                      storeCmd(YGCODE_UP, item_move_len[item_move_len_i]);
+                      storeCmd(y_axis_up, item_move_len[item_move_len_i]);
                       break;
                     case Z_AXIS:
-                      storeCmd(ZGCODE_UP, item_move_len[item_move_len_i]);
+                      storeCmd(z_axis_up, item_move_len[item_move_len_i]);
                       break;
                     default:
                       break;
@@ -265,13 +277,13 @@ void menuMove(void)
                     switch(a)
                     {
                     case X_AXIS:
-                      storeCmd(XGCODE_DOWN, item_move_len[item_move_len_i]);  
+                      storeCmd(x_axis_down, item_move_len[item_move_len_i]);  
                       break;
                     case Y_AXIS:
-                      storeCmd(YGCODE_DOWN, item_move_len[item_move_len_i]);
+                      storeCmd(y_axis_down, item_move_len[item_move_len_i]);
                       break;
                     case Z_AXIS:
-                      storeCmd(ZGCODE_DOWN, item_move_len[item_move_len_i]);
+                      storeCmd(z_axis_down, item_move_len[item_move_len_i]);
                       break;
                     default:
                       break;

--- a/TFT/src/User/Menu/Move.c
+++ b/TFT/src/User/Menu/Move.c
@@ -7,15 +7,10 @@
 #define X 0
 #define Y 1
 #define Z 2
-#define XGCODE_INC "G1 X%.1f\n"
-#define XGCODE_DEC "G1 X-%.1f\n"
-#define YGCODE_INC "G1 Y%.1f\n"
-#define YGCODE_DEC "G1 Y-%.1f\n"
-#define ZGCODE_INC "G1 Z%.1f\n"
-#define ZGCODE_DEC "G1 Z-%.1f\n"
 
-
-AXIS a = X_AXIS;
+#define X_MOVE_GCODE "G1 X%.1f\n"
+#define Y_MOVE_GCODE "G1 Y%.1f\n"
+#define Z_MOVE_GCODE "G1 Z%.1f\n"
 
 
 //1 title, ITEM_PER_PAGE item
@@ -46,6 +41,7 @@ LABEL_MOVE,
  }
 };
 
+const uint32_t item_move_speed[] = {DEFAULT_SPEED_MOVE, SPEED_MOVE_SLOW, SPEED_MOVE_FAST};
 
 #define ITEM_MOVE_LEN_NUM 3
 const ITEM itemMoveLen[ITEM_MOVE_LEN_NUM] = {
@@ -61,110 +57,65 @@ static u8    item_move_len_i = 1;
 static u32 nowTime = 0;
 static u32 update_time = 50; // 1 seconds is 100
 
+AXIS nowAxis = X_AXIS;
+
+void storeMoveCmd(AXIS xyz, int8_t direction) {
+  const char *xyzMoveCmd[] = {X_MOVE_GCODE, Y_MOVE_GCODE, Z_MOVE_GCODE};
+  // if invert is true, 'direction' multiplied by -1
+  storeCmd(xyzMoveCmd[xyz], (infoSettings.invert_axis[xyz] ? -direction : direction) * item_move_len[item_move_len_i]);
+  // update now axis be selected
+  nowAxis = xyz;
+}
+
+#define LOAD_XYZ_LABEL_INDEX(p0, dir0, p1, dir1, axis) do{ \
+                                                            moveItems.items[p0].label.index = LABEL_##axis##_##dir0; \
+                                                            moveItems.items[p1].label.index = LABEL_##axis##_##dir1; \
+                                                         }while(0)
+
 void menuMove(void)
 {
   KEY_VALUES  key_num = KEY_IDLE;
   
-  char * x_axis_up;
-  char * x_axis_down;
-  char * y_axis_up;
-  char * y_axis_down;
-  char * z_axis_up;
-  char * z_axis_down;
-  
+  // postion table of key
+  uint8_t table[TOTAL_AXIS][2] = 
   #ifdef ALTERNATIVE_MOVE_MENU
-      if(infoSettings.invert_xaxis == 1){
-        moveItems.items[4].label.index = LABEL_X_INC;
-        moveItems.items[6].label.index = LABEL_X_DEC;
-        x_axis_up   = XGCODE_DEC;
-        x_axis_down = XGCODE_INC;
-      }
-      else{
-        moveItems.items[4].label.index = LABEL_X_DEC;
-        moveItems.items[6].label.index = LABEL_X_INC;
-        x_axis_up   = XGCODE_INC;
-        x_axis_down = XGCODE_DEC;
-      }
-  
-      if(infoSettings.invert_yaxis == 1){
-        moveItems.items[1].label.index = LABEL_Y_DEC;
-        moveItems.items[5].label.index = LABEL_Y_INC;
-        y_axis_up   = YGCODE_DEC;
-        y_axis_down = YGCODE_INC;
-      }
-      else{
-        moveItems.items[1].label.index = LABEL_Y_INC;
-        moveItems.items[5].label.index = LABEL_Y_DEC;
-        y_axis_up   = YGCODE_INC;
-        y_axis_down = YGCODE_DEC;
-      }
-
-      if(infoSettings.invert_zaxis == 1){
-        moveItems.items[0].label.index = LABEL_Z_INC;
-        moveItems.items[2].label.index = LABEL_Z_DEC;
-        z_axis_up   = ZGCODE_INC;
-        z_axis_down = ZGCODE_DEC;
-      }
-      else{
-        moveItems.items[0].label.index = LABEL_Z_DEC;
-        moveItems.items[2].label.index = LABEL_Z_INC;
-        z_axis_up = ZGCODE_DEC;
-        z_axis_down = ZGCODE_INC;
-      }
+    /*-------*-------*-------*---------*
+     | Z-(0) | Y+(1) | Z+(2) | unit(3) |
+     *-------*-------*-------*---------*
+     | X-(4) | Y-(5) | X+(6) | back(7) |
+     *-------*-------*-------*---------*/
+    //X+ X-   Y+ Y-   Z+ Z-
+    {{6, 4}, {1, 5}, {2, 0}}
   #else
-      if(infoSettings.invert_xaxis == 1){
-        moveItems.items[0].label.index = LABEL_X_DEC;
-        moveItems.items[4].label.index = LABEL_X_INC;
-        x_axis_up   = XGCODE_DEC;
-        x_axis_down = XGCODE_INC;
-      }
-      else{
-        moveItems.items[0].label.index = LABEL_X_INC;
-        moveItems.items[4].label.index = LABEL_X_DEC;
-        x_axis_up   = XGCODE_INC;
-        x_axis_down = XGCODE_DEC;
-      }
-      if(infoSettings.invert_yaxis == 1){
-        moveItems.items[1].label.index = LABEL_Y_DEC;
-        moveItems.items[5].label.index = LABEL_Y_INC;
-        y_axis_up   = YGCODE_DEC;
-        y_axis_down = YGCODE_INC;
-      }
-      else{
-        moveItems.items[1].label.index = LABEL_Y_INC;
-        moveItems.items[5].label.index = LABEL_Y_DEC;
-        y_axis_up   = YGCODE_INC;
-        y_axis_down = YGCODE_DEC;
-      }
-        if(infoSettings.invert_zaxis == 1){
-        moveItems.items[2].label.index = LABEL_Z_DEC;
-        moveItems.items[6].label.index = LABEL_Z_INC;
-        z_axis_up   = ZGCODE_DEC;
-        z_axis_down = ZGCODE_INC;
-      }
-      else{
-        moveItems.items[2].label.index = LABEL_Z_INC;
-        moveItems.items[6].label.index = LABEL_Z_DEC;
-        z_axis_up   = ZGCODE_INC;
-        z_axis_down = ZGCODE_DEC;
-      }
+    /*-------*-------*-------*---------*
+     | X+(0) | Y+(1) | Z+(2) | unit(3) |
+     *-------*-------*-------*---------*
+     | X-(4) | Y-(5) | Z-(6) | back(7) |
+     *-------*-------*-------*---------*/
+    //X+ X-   Y+ Y-   Z+ Z-
+    {{0, 4}, {1, 5}, {2, 6}}
   #endif
+    ;
+  if(infoSettings.invert_axis[X_AXIS] != 1){
+    LOAD_XYZ_LABEL_INDEX(table[X_AXIS][0], INC, table[X_AXIS][1], DEC, X); // table[0] <--> INC(+) table[1] <--> DEC(+) if not inverted
+  } else {
+    LOAD_XYZ_LABEL_INDEX(table[X_AXIS][0], DEC, table[X_AXIS][1], INC, X); // table[0] <--> DEC(-) table[1] <--> INC(-) if inverted
+  }
+  if(infoSettings.invert_axis[Y_AXIS] != 1){
+    LOAD_XYZ_LABEL_INDEX(table[Y_AXIS][0], INC, table[Y_AXIS][1], DEC, Y);
+  } else {
+    LOAD_XYZ_LABEL_INDEX(table[Y_AXIS][0], DEC, table[Y_AXIS][1], INC, Y);
+  }
+  if(infoSettings.invert_axis[Z_AXIS] != 1){
+    LOAD_XYZ_LABEL_INDEX(table[Z_AXIS][0], INC, table[Z_AXIS][1], DEC, Z);
+  } else {
+    LOAD_XYZ_LABEL_INDEX(table[Z_AXIS][0], DEC, table[Z_AXIS][1], INC, Z);
+  }
 
   menuDrawPage(&moveItems);
   mustStoreCmd("G91\n");
+  mustStoreCmd("G1 F%d\n",item_move_speed[infoSettings.move_speed]);
 
-  switch (infoSettings.move_speed)
-  {
-  case 1:
-    mustStoreCmd("G1 F%d\n",SPEED_MOVE_SLOW);
-    break;
-  case 2:
-    mustStoreCmd("G1 F%d\n",SPEED_MOVE_FAST);
-    break;
-  default:
-    mustStoreCmd("G1 F%d\n",DEFAULT_SPEED_MOVE);
-    break;
-  }
   mustStoreCmd("M114\n");
   drawXYZ();
 
@@ -178,17 +129,9 @@ void menuMove(void)
     switch(key_num)
     {
       #ifdef ALTERNATIVE_MOVE_MENU
-        case KEY_ICON_0:
-              storeCmd(z_axis_up, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_1:
-              storeCmd(y_axis_up, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_2:
-              storeCmd(z_axis_down, item_move_len[item_move_len_i]);
-              break;
+        case KEY_ICON_0: storeMoveCmd(Z_AXIS, -1); break; // Z move down if no invert
+        case KEY_ICON_1: storeMoveCmd(Y_AXIS, 1); break;  // Y move increase if no invert
+        case KEY_ICON_2: storeMoveCmd(Z_AXIS, 1); break;  // Z move up if no invert
 
         case KEY_ICON_3:
               item_move_len_i = (item_move_len_i+1)%ITEM_MOVE_LEN_NUM;            
@@ -196,106 +139,38 @@ void menuMove(void)
               menuDrawItem(&moveItems.items[key_num], key_num);
               break;
 
-        case KEY_ICON_4:
-              storeCmd(x_axis_down, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_5:
-              storeCmd(y_axis_down, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_6:
-              storeCmd(x_axis_up, item_move_len[item_move_len_i]);
-              break;
+        case KEY_ICON_4: storeMoveCmd(X_AXIS, -1); break; // X move decrease if no invert
+        case KEY_ICON_5: storeMoveCmd(Y_AXIS, -1); break; // Y move decrease if no invert
+        case KEY_ICON_6: storeMoveCmd(X_AXIS, 1); break;  // X move increase if no invert
         
         case KEY_ICON_7: infoMenu.cur--; break;
-        default:break;
-
       #else
-        
-        case KEY_ICON_0:
-              a = X_AXIS;
-              storeCmd(x_axis_up, item_move_len[item_move_len_i]);
-              break;
-        
-        case KEY_ICON_1:
-              a = Y_AXIS;
-              storeCmd(y_axis_up, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_2:
-              a = Z_AXIS;
-              storeCmd(z_axis_up, item_move_len[item_move_len_i]);
-              break;
+        case KEY_ICON_0: storeMoveCmd(X_AXIS, 1); break;  // X move increase if no invert
+        case KEY_ICON_1: storeMoveCmd(Y_AXIS, 1); break;  // Y move increase if no invert
+        case KEY_ICON_2: storeMoveCmd(Z_AXIS, 1); break;  // Z move up if no invert
         
         case KEY_ICON_3:
               item_move_len_i = (item_move_len_i+1)%ITEM_MOVE_LEN_NUM;            
               moveItems.items[key_num] = itemMoveLen[item_move_len_i];
               menuDrawItem(&moveItems.items[key_num], key_num);
               break;
-            
-        case KEY_ICON_4:
-              a = X_AXIS;
-              storeCmd(x_axis_down, item_move_len[item_move_len_i]);
-              break;
 
-        case KEY_ICON_5:
-              a = Y_AXIS;
-              storeCmd(y_axis_down, item_move_len[item_move_len_i]);
-              break;
-
-        case KEY_ICON_6:
-              a = Z_AXIS;
-              storeCmd(z_axis_down, item_move_len[item_move_len_i]);
-              break;
+        case KEY_ICON_4: storeMoveCmd(X_AXIS, -1); break; // X move decrease if no invert
+        case KEY_ICON_5: storeMoveCmd(Y_AXIS, -1); break; // Y move decrease if no invert
+        case KEY_ICON_6: storeMoveCmd(Z_AXIS, -1); break; // Z move down if no invert
 
         case KEY_ICON_7: infoMenu.cur--; break;
-        
+      #endif        
         default:
-              #if LCD_ENCODER_SUPPORT
-                if(encoderPosition)
-                {
-                  if(encoderPosition > 0)
-                  {
-                    switch(a)
-                    {
-                    case X_AXIS:
-                      storeCmd(x_axis_up, item_move_len[item_move_len_i]);  
-                      break;
-                    case Y_AXIS:
-                      storeCmd(y_axis_up, item_move_len[item_move_len_i]);
-                      break;
-                    case Z_AXIS:
-                      storeCmd(z_axis_up, item_move_len[item_move_len_i]);
-                      break;
-                    default:
-                      break;
-                    }
-                  }
-                  if(encoderPosition < 0)
-                  {
-                    switch(a)
-                    {
-                    case X_AXIS:
-                      storeCmd(x_axis_down, item_move_len[item_move_len_i]);  
-                      break;
-                    case Y_AXIS:
-                      storeCmd(y_axis_down, item_move_len[item_move_len_i]);
-                      break;
-                    case Z_AXIS:
-                      storeCmd(z_axis_down, item_move_len[item_move_len_i]);
-                      break;
-                    default:
-                      break;
-                    }
-                  }
-                  encoderPosition = 0;    
-                }
-                LCD_LoopEncoder();
-              #endif
-              break;
-      #endif    
-
+          #if LCD_ENCODER_SUPPORT
+            if(encoderPosition)
+            {
+              storeMoveCmd(nowAxis, encoderPosition > 0 ? 1 : -1);             
+              encoderPosition = 0;    
+            }
+            LCD_LoopEncoder();
+          #endif
+          break;
     }
     loopProcess();
     update_gantry();
@@ -319,15 +194,15 @@ void drawXYZ(void){
   //GUI_SetColor(GANTRYLBL_BKCOLOR);
   //GUI_FillPrect(&RecXYZ);
   my_sprintf(tempstr, "X:%.1f  ", getAxisLocation(0));  
-  if (a == X_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
+  if (nowAxis == X_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
   GUI_DispString(START_X+1*SPACE_X+1*ICON_WIDTH,(ICON_START_Y-BYTE_HEIGHT)/2,(u8 *)tempstr);
   GUI_SetColor(FONT_COLOR);
   my_sprintf(tempstr, "Y:%.1f  ", getAxisLocation(1));
-  if (a == Y_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
+  if (nowAxis == Y_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
   GUI_DispString(START_X+2*SPACE_X+2*ICON_WIDTH,(ICON_START_Y-BYTE_HEIGHT)/2,(u8 *)tempstr);
   GUI_SetColor(FONT_COLOR);
   my_sprintf(tempstr, "Z:%.1f  ", getAxisLocation(2));
-  if (a == Z_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
+  if (nowAxis == Z_AXIS) GUI_SetColor(INFOBOX_ICON_COLOR);
   GUI_DispString(START_X+3*SPACE_X+3*ICON_WIDTH,(ICON_START_Y-BYTE_HEIGHT)/2,(u8 *)tempstr);
 
   //GUI_SetBkColor(BACKGROUND_COLOR);

--- a/TFT/src/User/Menu/Parametersetting.c
+++ b/TFT/src/User/Menu/Parametersetting.c
@@ -229,19 +229,19 @@ void parametersetting(){
         switch (key_num)
         {
         case KEY_ICON_0:
-            infoMenu.menu[++infoMenu.cur] = setStepsMenuItems;   break;
-        break;
+            infoMenu.menu[++infoMenu.cur] = setStepsMenuItems;
+            break;
 
         case KEY_ICON_1:
-            infoMenu.menu[++infoMenu.cur] = setCurrentMenuItems;   break;
-        break;
+            infoMenu.menu[++infoMenu.cur] = setCurrentMenuItems;
+            break;
 
         case KEY_ICON_7:
-        infoMenu.cur--;
-        break;
+            infoMenu.cur--;
+            break;
 
         default:
-        break;
+            break;
 
         }
 
@@ -303,12 +303,11 @@ float num_pad(float old_val)
         case NUM_KEY_EXIT:
             Send_Settingcmd();
             return old_val;
-            break;
 
         case NUM_KEY_DEL:
             if(nowIndex)
             ParameterBuf[--nowIndex] = 0;
-        break;
+            break;
 
         case NUM_KEY_1:
         case NUM_KEY_2:
@@ -357,7 +356,6 @@ float num_pad(float old_val)
 
             //Send_Settingcmd();
             return strtof(ParameterBuf,NULL);
-            break;
 
             default:
             break;

--- a/TFT/src/User/Menu/Settings.c
+++ b/TFT/src/User/Menu/Settings.c
@@ -16,11 +16,11 @@ void infoSettingsReset(void)
   infoSettings.silent = 0;
   infoSettings.auto_off = 0;
   infoSettings.terminalACK = 0;
-  infoSettings.invert_xaxis = 0;
-  infoSettings.invert_yaxis = 0;
+  infoSettings.invert_axis[X_AXIS] = 0;
+  infoSettings.invert_axis[Y_AXIS] = 0;
+  infoSettings.invert_axis[Z_AXIS] = 0;
   infoSettings.move_speed = 0;
   infoSettings.knob_led_color = (STARTUP_KNOB_LED_COLOR - 1); 
-  infoSettings.invert_zaxis = 0;
   infoSettings.send_start_gcode = 1;
   infoSettings.send_end_gcode = 1;
   infoSettings.persistent_info = 1;

--- a/TFT/src/User/Menu/Settings.h
+++ b/TFT/src/User/Menu/Settings.h
@@ -2,6 +2,7 @@
 #define _SETTINGS_H_
 
 #include "stdint.h"
+#include "coordinate.h"
 
 typedef enum
 {
@@ -22,11 +23,9 @@ typedef struct
   uint8_t  silent;
   uint8_t  auto_off;
   uint8_t  terminalACK;
-  uint8_t  invert_xaxis;
-  uint8_t  invert_yaxis;
+  uint8_t  invert_axis[TOTAL_AXIS];
   uint8_t  move_speed;
   uint8_t  knob_led_color;
-  uint8_t  invert_zaxis;
   uint8_t  send_start_gcode;
   uint8_t  send_end_gcode;
   uint8_t  persistent_info;

--- a/TFT/src/User/Menu/ledcolor.c
+++ b/TFT/src/User/Menu/ledcolor.c
@@ -31,7 +31,7 @@
                                               LED_VIOLET
                                               };
 
-void knob_LED_Init() // 12 11
+void knob_LED_Init(void) // 12 11
 {
     uint16_t psc = _PSC;
     uint16_t arr = _ARR;

--- a/TFT/src/User/Menu/ledcolor.h
+++ b/TFT/src/User/Menu/ledcolor.h
@@ -166,7 +166,7 @@
     extern const uint32_t led_color[LED_color_NUM];
 
 
-    void knob_LED_Init();
+    void knob_LED_Init(void);
     void knob_LED_DeInit(void);
     void ws2812_send_DAT(uint32_t ws2812_dat);
     #endif

--- a/TFT/src/User/main.c
+++ b/TFT/src/User/main.c
@@ -47,9 +47,7 @@ void Hardware_GenericInit(void)
   #endif
 
   #ifdef LED_color_PIN
-
     knob_LED_Init();
-
   #else
     #define STARTUP_KNOB_LED_COLOR 1
   #endif
@@ -65,7 +63,6 @@ void Hardware_GenericInit(void)
 
 int main(void)
 {
-
   SCB->VTOR = VECT_TAB_FLASH;
 
   Hardware_GenericInit();

